File: Cpp/Source/TextConverter.Test/source/HtmlPagesConverterTest.cpp

#include "TextConverter/include/HtmlPagesConverter.h"

#include <cppunit/TestCase.h>
#include <cppunit/TestFixture.h>
#include <cppunit/extensions/HelperMacros.h>

class HtmlPagesConverterTest : public CppUnit::TestFixture 
{
public:
    virtual void setUp() {}
    virtual void tearDown() {}

    void t_foo();
    
    CPPUNIT_TEST_SUITE(HtmlPagesConverterTest);
    CPPUNIT_TEST(t_foo);
    CPPUNIT_TEST_SUITE_END();
};

void HtmlPagesConverterTest::t_foo()
{
    HtmlPagesConverter converter("foo");
    CPPUNIT_ASSERT( std::string("fixme").compare( converter.getFilename() ) == 0 );
}

CPPUNIT_TEST_SUITE_REGISTRATION(HtmlPagesConverterTest);

File: php/tests/TextConverter/HtmlPagesConverterTest.php
<?php

declare(strict_types=1);

namespace Tests\TextConverter;

use PHPUnit\Framework\TestCase;
use RacingCar\TextConverter\HtmlPagesConverter;

class HtmlPagesConverterTest extends TestCase
{
    public function testFoo(): void
    {
        $converter = new HtmlPagesConverter('foo');
        $this->assertSame('fixme', $converter->getFileName());
    }
}
